### PR TITLE
🗃 Study file integration with Indexd

### LINKED ISSUE #55: Sync updates to Gen3
When relevant updates come in, they should be synchronously updated to Gen3 to provide authn/authz for the files.

### LINKED ISSUE #208: Integrate Study File with Gen3
The Study Files need to be registered and modified within Gen3/indexd in a similar fashion to Genomic Files.

### PR DESCRIPTION
**Pull request by dankolbman**
Update the study file model to use Indexd.

resolves #55 resolves #208

### CODE REVIEW

File: dataservice/api/study_file/resources.py [hunk 1]
```diff
@@ -33,10 +33,34 @@ def get(self, after, limit):
...
+            for st in pager.items:
+                merged = st.merge_indexd()
+                if merged is not None:
+                    keep.append(st)
+                else:
+                    refresh = True
+
+        # Replace original page's items with new list of valid files
+        pager.items = keep
+        pager.after = next_after if next_after else after
```
Comment (dankolbman) [inline dataservice/api/study_file/resources.py]: Need a better way of abstracting this from the resource

Comment (znatty22) [reply dataservice/api/study_file/resources.py]: Could you make this a method in flask_indexd and then do: 

```
from dataservice.extensions.flask_indexd import paginate as paginate_indexd
```

```
# Indexd Pagination
pager = paginate_indexd(q, after, limit)

return (StudyFileSchema(many=True)
                .jsonify(pager))
```

File: tests/study_file/test_study_file_resources.py [hunk 2]
```diff
@@ -11,114 +15,224 @@
...
+    st = StudyFile.query.get(kf_id)
+    st.merge_indexd()
+    assert st.file_name, body['file_name']
+    assert st.uuid == orig_uuid
+    assert st.latest_did != orig_did
+
+
+def test_delete(client, indexd, entities):
+    """
+    Test deleting a participant by id
```
Comment (znatty22) [inline tests/study_file/test_study_file_resources.py]: participant -> study_file

File: dataservice/api/study_file/resources.py (hunk 1 above)
Comment (dankolbman) [reply dataservice/api/study_file/resources.py]: Could do that, but maybe it's better lived in `common.pagination` or some sort of `IndexdResource`?

File: migrations/versions/83788dfcbd00_.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+"""empty message
```
Comment (znatty22) [inline migrations/versions/83788dfcbd00_.py]: Update doc string

**Review by znatty22 — COMMENTED** [review]
Looks good just one minor comment on updating docstring in migration. Then looks like its good to go

File: migrations/versions/83788dfcbd00_.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Create Date: 2018-04-24 09:20:50.988890
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '83788dfcbd00'
+down_revision = '243980a97e4c'
```
Comment (parimalak) [inline migrations/versions/83788dfcbd00_.py]: update down_revision to point to seq_center revision (4a7f7cb8ac00)

**Review by znatty22 — COMMENTED** [review]
👍

**Review by parimalak — APPROVED** [review]

**Pull request merged**